### PR TITLE
[Darc CLI] Implement simple get-subscriptions

### PR DESCRIPTION
**Pull request by mmitche**
Lists existing subscriptions, potentially filtered by source repo, target repo, channel, or target branch

### CODE REVIEW

**Comment by mmitche** [comment]
Part of #868

**Comment by mmitche** [comment]
This needs a slight tweak based on the merge policy stuff, stay tuned.

**Comment by mmitche** [comment]
Updated the formatting of the merge policies.  Now looks something like:

```
https://github.com/dotnet/arcade (.NET Tools - Latest) ==> 'https://github.com/dotnet/standard' ('master')                                   - Id: 13e4918d-b9bd-4474-29f1-08d6225f093d                                                                                                 - Update Frequency: everyDay                                                                                                               - Merge Policies:                                                                                                                            AllChecksSuccessful                                                                                                                          ignoreChecks =                                                                                                                                            [                                                                                                                                            "WIP",                                                                                                                                     "license/cla"                                                                                                                            ]                                                                                                                       - Last Build: N/A                                                                                                                        
```

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionsOperation.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                        subscription.Channel.Name.Contains(_options.Channel, StringComparison.OrdinalIgnoreCase));
+                    });
+
+                if (subscriptions.Count() == 0)
+                {
+                    Console.WriteLine("No subscriptions found matching the specified criteria");
+
+                }
+
+                // Based on the current output scheme, sort by source repo, target repo, target branch, etc.
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionsOperation.cs]: >scheme [](start = 47, length = 6)

scheme -> schema

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionsOperation.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    class GetSubscriptionsOperation : Operation
+    {
+        GetSubscriptionsComandLineOptions _options;
+        public GetSubscriptionsOperation(GetSubscriptionsComandLineOptions options)
+            : base(options)
+        {
+            _options = options;
+        }
+
+        public override int Execute()
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionsOperation.cs]: >override in [](start = 15, length = 11)

In your other PR Execute became async probably you want to change now or merge that one and then update this PR

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionsOperation.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                                foreach (string line in valueLines)
+                                {
+                                    Console.WriteLine($"{indentString}{line}");
+                                }
+                            }
+                        }
+                    }
+                    Console.WriteLine($"  - Last Build: {(subscription.LastAppliedBuild != null ? subscription.LastAppliedBuild.BuildNumber : "N/A")}");
+                }
+                return 0;
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionsOperation.cs]: >0 [](start = 23, length = 1)

Use Constants.SuccessCode (depending on the merge order)

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionsOperation.cs (hunk 2 above)
Comment (mmitche) [reply src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionsOperation.cs]: Going to merge this one first.

**Pull request merged**